uint16[] chan_1
uint16[] chan_2
